src/components/CascadeMenu/PCascadePanels.vue

```diff
@@ -0,0 +1,88 @@
+<template>
+  <div v-show="isOpen" class="p-cascade-panels">
+    <template v-for="{ id } in panels" :key="id">
+      <transition name="p-cascade-panels__panel-transition" tag="keep-alive">
+        <PCascadePanel v-show="panelIsOpen(id)" :panel-id="id" class="p-cascade-panels__panel">
+          <template #default="panel">
+            <slot :name="`${getBaseSlotName(id)}`" v-bind="panel">
+              {{ getBaseSlotName(id) }}
+            </slot>
+          </template>
+        </PCascadePanel>
+      </transition>
+    </template>
+  </div>
+</template>
+
+<script setup lang="ts">
+  import { inject } from 'vue'
+  import { PCascadePanel } from '@/components'
+  import { CascadePanel, CascadePanelId, cascadePanelsKey, useCascadePanels } from '@/compositions'
+  import { kebabCase } from '@/utilities'
+
+  const props = defineProps<{
+    panels: CascadePanel[],
+  }>()
+
+  // Calling inject with a nullish coallescing operator while providing a default value of
+  // undefined allows callers to use the `useCascadePanels` composition function while
+  // ensuring that the composition state is the same.
+  const {
+    close,
+    closeAll,
+    closePanelById,
+    getPanelById,
+    isOpen,
+    open,
+    openPanelById,
+    openPanels,
+    panelIsOpen,
+    toggle,
+    togglePanelById,
+  } = inject(cascadePanelsKey, undefined) ?? useCascadePanels(() => props.panels)
```

Comment:
It's more a devex piece so that you can do something like:
```vue
<template>
  <p-cascade-panels :panels />
</template>

<script lang="ts" setup>
import { useCascadePanels } from '@prefecthq/prefect-design'

const panels = [/* whatever */]

const { open, openPanelById, ...rest } = useCascadePanels(panels)
</script>
```
without needing to bind the component like we do elsewhere:

```vue
<template>
  <p-cascade-panels ref="cascadePanels" :panels />
</template>

<script lang="ts" setup>
import { ref } from 'vue'
import { PCascadePanels } from '@prefecthq/prefect-design'

const panels = [/* whatever */]

const cascadePanels = ref<typeof PCascadePanels>()
// now we can use the same methods via cascadePanels.value.method but we need to check
// everywhere that cascadePanels.value exists

</script>

Reply:
So if `useCascadePanels` has already been called at a higher level that same context will be used in the panels component otherwise it creates its own context?

Reply:
Correct!

Reply:
Though I wonder if we might get into trouble with that and this needs to check if the panels are equivalent

Reply:
Actually yes we do need to check that lol

Reply:
lol I trust your instincts here, just wanted to make sure I understood the intent

Reply:
Ok I added some logic to handle injection scenarios like this; it's not the prettiest but I think the devX is nicer for it